### PR TITLE
update project validation

### PR DESCRIPTION
**Pull request by AnnPostnikova**
http://outsourcing.nat.tepkom.ru:20080/redmine/issues/1017

### CODE REVIEW

File: src/ProjectService.Validation/Image/CreateImagesRequestValidator.cs [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+{
+  public class CreateImagesRequestValidator : AbstractValidator<CreateImagesRequest>, ICreateImagesRequestValidator
+  {
+    public CreateImagesRequestValidator(
+      IImageValidator imageValidator,
+      IProjectRepository projectRepository)
+    {
+      RuleFor(images => images.Images)
+        .Cascade(CascadeMode.Stop)
+        .NotNull().WithMessage("List must not be null.")
```
Comment (MariyaSin) [inline src/ProjectService.Validation/Image/CreateImagesRequestValidator.cs:16]: It's not entirely clear what's wrong
better "List of images must not be null."

File: src/ProjectService.Validation/Image/CreateImagesRequestValidator.cs [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+        .NotEmpty().WithMessage("List must not be empty.")
```
Comment (MariyaSin) [inline src/ProjectService.Validation/Image/CreateImagesRequestValidator.cs:17]: same

File: src/ProjectService.Validation/Image/CreateImagesRequestValidator.cs [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+        .ForEach(image =>
+        {
+          image
+          .Cascade(CascadeMode.Stop)
+          .NotNull().WithMessage("Image content must not be null.")
+          .SetValidator(imageValidator);
+        });
+
+      RuleFor(images => images.ProjectId)
```
Comment (MariyaSin) [inline src/ProjectService.Validation/Image/CreateImagesRequestValidator.cs:26]: same

File: src/ProjectService.Validation/Image/CreateImagesRequestValidator.cs [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+        .Cascade(CascadeMode.Stop)
+        .NotEmpty().WithMessage("Project id must not be empty.")
+        .MustAsync(async (x, _) => await projectRepository.DoesExistAsync(x))
+        .WithMessage("Project id is invalid");
```
Comment (MariyaSin) [inline src/ProjectService.Validation/Image/CreateImagesRequestValidator.cs:30]: "Invalid project id."

File: src/ProjectService.Validation/Image/RemoveImagesRequestValidator.cs [hunk 5]
```diff
@@ -4,14 +4,15 @@
...
 namespace LT.DigitalOffice.ProjectService.Validation.Image
 {
-  public class RemoveImageValidator : AbstractValidator<RemoveImageRequest>, IRemoveImageValidator
+  public class RemoveImagesRequestValidator : AbstractValidator<RemoveImageRequest>, IRemoveImagesRequestValidator
   {
-    public RemoveImageValidator()
+    public RemoveImagesRequestValidator()
     {
       RuleFor(list => list.ImagesIds)
         .NotNull().WithMessage("List must not be null.")
```
Comment (MariyaSin) [inline src/ProjectService.Validation/Image/RemoveImagesRequestValidator.cs:12]: same

File: src/ProjectService.Validation/Image/RemoveImagesRequestValidator.cs [hunk 6]
```diff
@@ -4,14 +4,15 @@
...
         .NotEmpty().WithMessage("List must not be empty.")
```
Comment (MariyaSin) [inline src/ProjectService.Validation/Image/RemoveImagesRequestValidator.cs:13]: same

File: src/ProjectService.Business/Commands/Image/CreateImageCommand.cs [hunk 7]
```diff
@@ -103,16 +108,15 @@ public async Task<OperationResultResponse<List<Guid>>> ExecuteAsync(CreateImages
...
-
-        return new OperationResultResponse<List<Guid>>
-        {
-          Status = OperationResultStatusType.Failed,
-          Errors = errors
-        };
+        return _responseCreator.CreateFailureResponse<List<Guid>>(
+          HttpStatusCode.BadRequest,
+          validationResult.Errors.Select(x => x.ErrorMessage)
+          .ToList());
```
[the remark this answers is not in the document]
Comment (AnnPostnikova) [reply src/ProjectService.Business/Commands/Image/CreateImageCommand.cs]: done

File: src/ProjectService.Validation/Image/CreateImagesRequestValidator.cs (hunk 1 above)
Comment (AnnPostnikova) [reply src/ProjectService.Validation/Image/CreateImagesRequestValidator.cs]: done

File: src/ProjectService.Validation/Image/CreateImagesRequestValidator.cs (hunk 3 above)
Comment (AnnPostnikova) [reply src/ProjectService.Validation/Image/CreateImagesRequestValidator.cs]: done

File: src/ProjectService.Validation/Image/RemoveImagesRequestValidator.cs (hunk 5 above)
Comment (AnnPostnikova) [reply src/ProjectService.Validation/Image/RemoveImagesRequestValidator.cs]: done

File: src/ProjectService.Validation/Image/CreateImagesRequestValidator.cs [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+    {
+      RuleFor(request => request.Images)
+        .Cascade(CascadeMode.Stop)
+        .NotNull().WithMessage("List of images must not be null.")
+        .NotEmpty().WithMessage("List of images must not be empty.")
+        .ForEach(image =>
+        {
+          image
+          .Cascade(CascadeMode.Stop)
+          .NotNull().WithMessage("Image content must not be null.")
```
Comment (ndoriomedov) [inline src/ProjectService.Validation/Image/CreateImagesRequestValidator.cs:22]: maybe remove content from the error message?

**Review by ndoriomedov — APPROVED** [review]

**Review by AnnaGakh — APPROVED** [review]

**Review by vorobevamv — APPROVED** [review]

**Review by a5x55555 — APPROVED** [review]

**Pull request merged**